frontend/src/components/floating-menus/MenuList.vue

```diff
@@ -6,44 +6,51 @@
 		:type="'Dropdown'"
 		:windowEdgeMargin="0"
 		:escapeCloses="false"
-		v-bind="{ direction, scrollableY, minWidth }"
+		v-bind="{ direction, minWidth, scrollableY: !fixedHeight && scrollableY }"
 		ref="floatingMenu"
 		data-hover-menu-keep-open
 	>
-		<template v-for="(section, sectionIndex) in entries" :key="sectionIndex">
-			<Separator :type="'List'" :direction="'Vertical'" v-if="sectionIndex > 0" />
-			<LayoutRow
-				v-for="(entry, entryIndex) in section"
-				:key="entryIndex"
-				class="row"
-				:class="{ open: isEntryOpen(entry), active: entry.label === highlighted?.label }"
-				@click="() => onEntryClick(entry)"
-				@pointerenter="() => onEntryPointerEnter(entry)"
-				@pointerleave="() => onEntryPointerLeave(entry)"
-			>
-				<CheckboxInput v-if="entry.checkbox" v-model:checked="entry.checked" :outlineStyle="true" :disableTabIndex="true" class="entry-checkbox" />
-				<IconLabel v-else-if="entry.icon && drawIcon" :icon="entry.icon" class="entry-icon" />
-				<div v-else-if="drawIcon" class="no-icon"></div>
-
-				<span class="entry-label">{{ entry.label }}</span>
-
-				<IconLabel v-if="entry.shortcutRequiresLock && !fullscreen.state.keyboardLocked" :icon="'Info'" :title="keyboardLockInfoMessage" />
-				<UserInputLabel v-else-if="entry.shortcut?.length" :inputKeys="[entry.shortcut]" />
-
-				<div class="submenu-arrow" v-if="entry.children?.length"></div>
-				<div class="no-submenu-arrow" v-else></div>
-
-				<MenuList
-					v-if="entry.children"
-					@naturalWidth="(newNaturalWidth: number) => $emit('naturalWidth', newNaturalWidth)"
-					:open="entry.ref?.open || false"
-					:direction="'TopRight'"
-					:entries="entry.children"
-					v-bind="{ defaultAction, minWidth, drawIcon, scrollableY }"
-					:ref="(ref: typeof FloatingMenu) => ref && (entry.ref = ref)"
-				/>
-			</LayoutRow>
-		</template>
+		<LayoutCol ref="scroller" :scrollableY="scrollableY && fixedHeight" @scroll="onScroll" :style="{ minWidth: fixedHeight ? `${minWidth}px` : `inherit` }">
+			<LayoutRow v-if="fixedHeight" class="spacer" :style="{ height: `${startIndex * fixedHeight}px` }"></LayoutRow>
```

Comment:
Can you please rename the `fixedHeight` prop to `virtualScrolling` and the class `spacer` to `virtual-scrolling-spacer`?

Reply:
Wait, sorry, I didn't realize `fixedHeight` is a number not a bool. Maybe `virtualScrollingTotalHeight`?